gunicorn/arbiter.py

```diff
@@ -636,11 +640,5 @@ def kill_worker(self, pid, sig):
             os.kill(pid, sig)
         except OSError as e:
             if e.errno == errno.ESRCH:
-                try:
```

Comment:
why removing it?

Reply:
It duplicates code in `reap_workers`. If we can guarantee we don't miss the CHLD signal, then we will not need this.

Reply:
Also, this is called from `murder_workers`. By removing this code, we remove the concurrent modification and looping, and we no longer need to call `list()`.